logger/httplog/round_trip_stringer.go

```diff
@@ -69,15 +69,17 @@ func (r RoundTripStringer) String() string {
 		sb.WriteString("\n")
 	}
 	sb.WriteString("< ")
-	if r.Response != nil {
-		sb.WriteString(fmt.Sprintf("%s %s", r.Response.Proto, r.Response.Status))
-		// Only display error on this line if the response body is empty.
-		// Otherwise the response body will include details about the error.
-		if len(r.ResponseBody) == 0 && r.Err != nil {
-			sb.WriteString(fmt.Sprintf(" (Error: %s)", r.Err))
-		}
-	} else {
+	if r.Response == nil {
 		sb.WriteString(fmt.Sprintf("Error: %s", r.Err))
+		return sb.String()
+	}
+
+	sb.WriteString(fmt.Sprintf("%s %s", r.Response.Proto, r.Response.Status))
+	// Only display error on this line if the response body is empty or the
+	// client failed to read the response body.
+	// Otherwise the response body will include details about the error.
+	if len(r.ResponseBody) == 0 && r.Err != nil {
+		sb.WriteString(fmt.Sprintf(" (Error: %s)", r.Err))
```

Comment:
a question, when length of response body is > 0 (line 86), is it always true that the Err would be nil?

Reply:
Not necessarily. We call the ErrorMapper function that parses the response into one of the SDK errors depending on the result. If the API responds with any of our standard error codes, `r.Err` will be set to that.

This is just refactored from above, so there is no behavior change here.